### ISSUE TITLE
clicking on timestamp doesn't focus on video

### ISSUE THREAD

**Issue by hcgiub001** [pc035860/YCS-cont#11]
Can we have it when we click on timestamp that the video gets focus. This behaviour is normal with using general comments in YouTube but not in YCS. thanks for consideration

**Comment by NewsGuyTor** [comment]
I don't want this, at all! If this is implemented, it should be an optional opt-in setting, not something default. 

This would totally destroy the extension user experience while watching videos in PiP.

**Comment by hcgiub001** [comment]
> I don't want this, at all! If this is implemented, it should be an optional opt-in setting, not something default.
> 
> This would totally destroy the extension user experience while watching videos in PiP.

I didn't realise this could cause an issue with pip. have you tested with normal timeline links to see if it affects it? if it does then allowing it as an option would be cool

**Comment by hcgiub001** [comment]
I'm not a coder but if it can detect it's in pip then you can alter the behaviour so it takes focus if not in pip and doesn't take focus if in pip

**Comment by NewsGuyTor** [comment]
Yeah, it's an issue with normal comments, it scrolls to the top, massively annoying! I converted this [extension](https://chromewebstore.google.com/detail/prevent-timestamp-scrolli/khcmbjhfmcdgeobfonbjhibjpaaembla?hl=en) to an userscript that fixes it here:

https://gist.github.com/NewsGuyTor/ff655400ff400d192de7e83759386971

BTW, I recommend that you install an embedded PIP extension (like [Enhancer](https://www.mrfdev.com/enhancer-for-youtube)) installed instead, I don't get why you'd ever prefer to lose context of where you are in a comment thread. It's a much more pleasant experience (vid has no sound):

https://github.com/pc035860/YCS-cont/assets/19603024/fec39f6e-4cbc-452d-8b13-f0db9eae22b8

**Comment by drhouse** [comment]
I can highly recommend this chrome addon that shows timestamped comments at the appropriate time on the upper-corner of the YouTube video: [Timestamped YouTube Comments](https://chromewebstore.google.com/detail/timestamped-youtube-comme/gnpppldhfbbagiaalkddddajadhlgofm)
<img width="640" alt="unnamed" src="https://github.com/pc035860/YCS-cont/assets/187476/2af3a6d9-5e26-431d-82e5-74eee5272d8c">

**Features:**
- Pause to persist a comment
- Hover over a comment to see its replies
- Click the Comments icon in the video bar to toggle comments on/off 
- Click on a comment to open a new tab with that comment at the top of the comment section (useful for replying to comments)

**Comment by hcgiub001** [comment]
> Yeah, it's an issue with normal comments, it scrolls to the top, massively annoying! I converted this [extension](https://chromewebstore.google.com/detail/prevent-timestamp-scrolli/khcmbjhfmcdgeobfonbjhibjpaaembla?hl=en) to an userscript that fixes it here: https://gist.github.com/NewsGuyTor/ff655400ff400d192de7e83759386971
> 
> BTW, I recommend that you install an embedded PIP extension (like [Enhancer](https://www.mrfdev.com/enhancer-for-youtube)) instead, I don't get why you'd ever prefer to lose context of where you are in a comment thread. It's a much more pleasant experience (vid has no sound):
> 
>  ScreenFlow.mp4

I tested myself and can see how annoying this is. But can you also see how annoying it could be for those that don't use pip?
I've used enhancer before for that pip functionality. I found I liked video being paused when viewing comments.

but I found a nice extension that "Comments - Scroll to comment (after clicking on a timestamp or changing fullscreen mode)"
it's called Tweaks, it's a bloody amazing extension. you can set a shortcut for that action. so I click on a timestamp. video takes focus. then I can click the shortcut so it goes exactly back to where I clicked the timeline.

If you don't use pip, can you see I have to scroll up to watch the video and I lose focus on the comment anyway

**Comment by hcgiub001** [comment]
> I can highly recommend this chrome addon that shows timestamped comments at the appropriate time on the upper-corner of the YouTube video: [Timestamped YouTube Comments](https://chromewebstore.google.com/detail/timestamped-youtube-comme/gnpppldhfbbagiaalkddddajadhlgofm) <img alt="unnamed" width="640" src="https://private-user-images.githubusercontent.com/187476/338486824-2af3a6d9-5e26-431d-82e5-74eee5272d8c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMDQwNjQsIm5iZiI6MTcxODEwMzc2NCwicGF0aCI6Ii8xODc0NzYvMzM4NDg2ODI0LTJhZjNhNmQ5LTVlMjYtNDMxZC04MmU1LTc0ZWVlNTI3MmQ4Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxMVQxMTAyNDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMDhlOWQyMThjYmUyOGEyMzMzY2EyNmU4ZmQ1N2RmNTQ2MDIyNTdmZWQ2NTFhMzdhNDk4MjZmYzU1ODY4ZmRmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.oytNYfuCv7Qbxv1sP8Z0k71BWULiejsbvKzGCMUjtxA">
> 
> **Features:**
> 
> * Pause to persist a comment
> * Hover over a comment to see its replies
> * Click the Comments icon in the video bar to toggle comments on/off
> * Click on a comment to open a new tab with that comment at the top of the comment section (useful for replying to comments)

that's actually cool thanks

**Comment by hcgiub001** [comment]
@NewsGuyTor if you want more information on tweaks. let me know.
